lib/puppet/node.rb

```diff
@@ -27,7 +27,10 @@ def initialize_from_hash(data)
     @name       = data['name']       || (raise ArgumentError, _("No name provided in serialized data"))
     @classes    = data['classes']    || []
     @parameters = data['parameters'] || {}
-    @environment_name = data['environment']
+    env_name = data['environment']
+    env_name = env_name.intern unless env_name.nil?
+    @environment_name = env_name
+    environment = env_name
```

Comment:
The `puppet apply` case seems to be covered by the acceptance tests _environment/enc_nonexistent_directory_environment.rb_ and _environment/environment-scenario-non_exists.rb_ (covers config, module_install, and apply). Those tests pass. I don't think `puppet resource` does anything differently that would affect this ticket, but if it does, then CI should add a test for that case.